controllers/operatorpolicy_controller.go

```diff
@@ -1623,9 +1629,26 @@ func (r *OperatorPolicyReconciler) handleCatalogSource(
 		CatalogSrcState := catalogSrc.Status.GRPCConnectionState.LastObservedState
 		isUnhealthy = (CatalogSrcState != CatalogSourceReady)
 	}
+	
+	if policy.Spec.ComplianceType.IsMustHave() {
+		changed := updateStatus(policy, catalogSourceFindCond(isUnhealthy, isMissing, catalogName),
+			catalogSourceObj(catalogName, catalogNS, isUnhealthy, isMissing))
+
+		return changed, nil
+	}
+
+	// Mustnothave
+	var relObj policyv1.RelatedObject
+	cond := notApplicableCond("CatalogSource")
+	cond.Status = metav1.ConditionFalse // CatalogSource condition has the opposite polarity
+
+	if isMissing {
+		relObj = missingNotApplicableObj(catalogName, catalogNS, policyv1.MustNotHave, catalogSrcGVK)
+	} else {
+		relObj = foundNotApplicableObj(catalogSrc)
+	}
 
-	changed := updateStatus(policy, catalogSourceFindCond(isUnhealthy, isMissing, catalogName),
-		catalogSourceObj(catalogName, catalogNS, isUnhealthy, isMissing))
+	changed := updateStatus(policy, cond, relObj)
```

Comment:
Since mustnothave mode does not need to check if the catalogsource is healthy, can it be handled earlier? I have a feeling it would clean up some of this. Right now, in particular, I needed to double-check that the `catalogSrc` here wasn't a nil pointer - but I think it could've just used `foundCatalogSrc`?